### ISSUE TITLE
Random seed

### ISSUE THREAD

**Issue by TannerJ07** [david-pfx/PuzzleScriptNext#92]
Check ([ ] -> [x]) all that apply, then describe the issue below:

- [x] I have searched the tracker for similar issues (including closed ones)
- [x] I have provided a detailed example for how this feature should work
- [x] I have provided a use case for this feature, e.g. short description/share link of a game you're making and how you'd use the feature in that game, other things you've tried, edge cases, etc. (if applicable)

Random seed would work exactly like random, but it uses the current state of the level to generate the next random result. This would mean that undoing and then moving the same way would have the same result.

Just to be more specific: I imagine the current random uses a random number generator and chooses based on that. Random seed would use a random number generator based on the current state, and then it would choose a rule.

Best use case: any random game where you want undoing allowed, but still randomness that cant be controlled by repeating the same move.

**Comment by david-pfx** [comment]
Yes, a feature to do that is a good idea. I was astonished to find out that undo/restart do not work that way.

I think my preference is to have a prelude setting and to track the seed on the undo stack. It looks simpler, both to use and to implement, but I will have to check whether the inbuilt RNG is friendly to that kind of thing.

**Comment by david-pfx** [comment]
BTW my 2048 game is a good example. You can always beat it by undoing a few times.
